api/v1/inferencepool_types.go

```diff
@@ -95,26 +100,26 @@ type Extension struct {
 	//
 	// +optional
 	// +kubebuilder:default=Service
-	Kind *Kind `json:"kind,omitempty"`
+	Kind Kind `json:"kind,omitempty"`
```

Comment:
Why do we change it to be non-pointer while the previous `Group` is still a pointer?
Kind of got confused about the pointer and non-pointer choice with https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required?
@rikatz